### DISCUSSION TITLE
Temporary upload translation bug

### DISCUSSION THREAD

**Discussion by Yinci** [livewire/livewire#4644] [7. Bugs (archived)]
In the search for an answer, I stumbled upon #1546. This issue is closed, however it seems the issue is not resolved. As described, the temporary upload validation does not respect the attribute translation. 

I have been digging through the source code and the issue seems to be related to the `src/WithFileUploads::uploadError()` method, where the attribute name does not get translated in the errors returned. I would open a PR, but with my limited understanding I wouldn't be certain if this is the core of the issue (or even the best place to resolve it), but basically the `files.0` doesn't properly translate in the error message returned in the `$errorsInJson` variable, instead just gets replaced with the passed attribute name. I did try using the `$validationAttributes` property, but that is never even requested.

P.S. In my experience the code on line 58 and 59 (which, in this case doesn't get processed, but could still be relevant for the issue to be resolved), using `$translator->get()` doesn't work if the attribute contains dots. Therefore I would recommend a change to line 59:

```
if ($attribute === "validation.attributes.{$name}") {
    $attribute = $translator->get('validation.attributes')[$name] ?? $name;
}
```

If there is already a fix for this, if I'm just misunderstanding something, or if this is a duplicate, then I apologize in advance.

**Versions:**
Laravel: 8.80.0
Livewire: 2.9.0

**Comment by joshhanley** [comment]
@Yinci yeah I think you are right, I do remember there being an issue with this.

I think it's here somewhere, where it should translate but it doesn't https://github.com/livewire/livewire/blob/2d68c61a8edf338534fdd8e2b2750dca2e741439/src/Testing/Concerns/MakesCallsToComponent.php#L112

Where as you can see for the main validation it has messages and attributes properties passed in https://github.com/livewire/livewire/blob/0a878717bec8cd94aa3b30bccb905eaeef98dfd2/src/ComponentConcerns/ValidatesInput.php#L178

Give a PR a go if you can! Hopefully this might help you with it!

**Comment by marchershey** [comment]
I'm pulling this out of the grave, because I'm currently having the same issue. Many times I've found that Caleb has fixed this issue, but I'm still seeing it on my side, using V3. 

See the following:
- https://github.com/livewire/livewire/issues/1546
- https://github.com/livewire/livewire/issues/1524
- Where Caleb fixed the issue? https://github.com/livewire/livewire/commit/c8d1e3771badb8e76fcf40d8fbc394f97d57a359

So was this resolved? If so, what am I doing wrong?

**Please note:** I am using Flux, maybe this issue was reintroduced? 

```html
<!-- index.blade.php -->
<flux:input type="file" wire:model.live="temp_photos" ... />
```

```php
// Controller.php
// (please ignore all the extras, I was trying everything lol
#[Validate([
    'temp_photos' => 'nullable',
    'temp_photos.*' => 'image|mimetypes:image/jpeg,image/jpg,image/png,image/webp,image/bmp|mimes:jpg,jpeg,png,webp,bmp|extensions:jpg,jpeg,png,webp,bmp|max:12288',
], message: [
    'temp_photos.*.image' => 'Only image files are permitted for upload.',
    'temp_photos.*.max' => 'Uploaded images can not exceed a max file size of 12MB.',
    'temp_photos.*.mimes' => '1Images must be in a common image format. (e.g., jpeg, jpg, png, webp, and bmp)',
    'temp_photos.*.extensions' => '2Images must be in a common image format. (e.g., jpeg, jpg, png, webp, and bmp)',
], as: [
    'temp_photos' => 'selected photos',
    'temp_photos.*' => 'selected photo',
], attribute: [
    'temp_photos' => 'selected photos',
    'temp_photos.*' => 'selected photo',
])]
public $temp_photos = [];
```
